### PR TITLE
[IMP] mail: restore follow/unfollow button

### PR DESCRIPTION
**Pull request by seb-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/views/chatter.js [hunk 1]
```diff
@@ -10,16 +10,17 @@ import { useService } from "@web/core/utils/hooks";
 
 export class Chatter extends Component {
     setup() {
-        this.messaging = useMessaging();
+        this.action = useService("action");
         this.activity = useService("mail.activity");
+        this.messaging = useMessaging();
+        this.orm = useService("orm");
```
Comment (phenix-factory) [inline addons/mail/static/src/new/views/chatter.js:16]: orm service is already exposed by messaging via `this.messaging.orm`

File: addons/mail/static/src/new/views/chatter.js [hunk 2]
```diff
@@ -10,16 +10,17 @@ import { useService } from "@web/core/utils/hooks";
...
         this.rpc = useService("rpc");
-        this.action = useService("action");
+        this.userService = useService("user");
```
Comment (phenix-factory) [inline addons/mail/static/src/new/views/chatter.js]: Look unused ?

File: addons/mail/static/src/new/views/chatter.js (hunk 1 above)
Comment (ged-odoo) [reply addons/mail/static/src/new/views/chatter.js:16]: it's not equivalent. with useService, the orm methods are "protected" => they  will stay pending if composer is destroyed, but if you take it from messaging, you no longer benefit from that. I think we should always use the service from useService

Comment (phenix-factory) [reply addons/mail/static/src/new/views/chatter.js:16]: Good to know then !

File: addons/mail/static/src/new/views/chatter.js (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/new/views/chatter.js]: good catch

File: addons/mail/static/src/new/views/chatter.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/new/views/chatter.js:16]: There are a lot of cases where don't want to protect them though (https://github.com/odoo-dev/odoo/pull/1448). I will keep it here, but I'm not so found of the feature, I think it's too magic and too hard to debug when the protection was not desired.

**Review by phenix-factory — APPROVED** [review]

**Review by alexkuhn — APPROVED** [review]

**Pull request merged**